src/SAWScript/Proof.hs

```diff
@@ -288,13 +461,288 @@ trivialProofTerm sc (Prop p) = runExceptT (loop =<< lift (scWhnf sc p))
                , showTerm p
                ]
 
+normalizeProp :: SharedContext -> ModuleMap -> Set VarIndex -> Prop -> IO Prop
+normalizeProp sc modmap opaqueSet (Prop tm) =
+  do tm' <- TM.normalizeSharedTerm sc modmap mempty mempty opaqueSet tm
+     termToProp sc tm'
+
 -- | Pretty print the given proposition as a string.
-prettyProp :: PPOpts -> Prop -> String
-prettyProp opts (Prop tm) = scPrettyTerm opts tm
+prettyProp :: PPOpts -> SAWNamingEnv -> Prop -> String
+prettyProp opts nenv p = renderSawDoc opts (ppProp opts nenv p)
 
 -- | Pretty print the given proposition as a @SawDoc@.
-ppProp :: PPOpts -> Prop -> SawDoc
-ppProp opts (Prop tm) = ppTerm opts tm
+ppProp :: PPOpts -> SAWNamingEnv -> Prop -> SawDoc
+ppProp opts nenv (Prop tm) = ppTermWithNames opts nenv tm
+
+-- TODO, I'd like to add metadata here
+type SequentBranch = Prop
+
+data FocusedBranch = FB ![SequentBranch] !SequentBranch ![SequentBranch]
+
+data Sequent
+  = UnfocusedSequent   ![SequentBranch] ![SequentBranch]
+  | GoalFocusedSequent ![SequentBranch] !FocusedBranch
```

Comment:
I vaguely recall you mentioning that you want to prefer calling these "conclusions" rather than "goals". Is that change of terminology something that you are still planning to do?

Reply:
Yeah, I think I will do that.